src/Console/Shell.php

```diff
@@ -307,6 +310,7 @@ public function dispatchShell()
         if (is_string($args[0]) && count($args) === 1) {
             $args = explode(' ', $args[0]);
         }
+        $args[] = '--requested';
```

Comment:
I wonder if we could skip adding a new cli flag to all shells if instead we supported 'extra' parameters like requestAction() does. It might get a bit twisty though as ShellDispatcher and Shell would have to know about these new extra parameters.